org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/references/CompoundTypeReference.java

```diff
@@ -113,6 +113,15 @@ public boolean isArray() {
 	}
 	
 	@Override
+	public LightweightTypeReference getComponentType() {
+		for(LightweightTypeReference component: expose(components)) {
+			if (component.isArray())
+				return component.getComponentType();
+		}
+		throw new IllegalStateException("CompoundTypeReference is an array but none of its components is an array.");
```

Comment:
I don't see a test for this#isArray(). I'd prefer to simply cal super and use the general contract to return null.